src/Microsoft.SqlTools.ServiceLayer/LanguageServices/PeekDefinition.cs

```diff
@@ -0,0 +1,210 @@
+        {
+            return (schemaName != null) ? database.Tables[tableName, schemaName]?.Script()
+                    : database.Tables[tableName].Script();
+        }
+
+        /// <summary>
+        /// Script a view using SMO
+        /// </summary>
+        /// <param name="viewName">View name</param>
+        /// <param name="schemaName">Schema name </param>
+        /// <returns>String collection of scripts</returns>
+        internal StringCollection GetViewScripts(string viewName, string schemaName)
+        {
+            return (schemaName != null) ? database.Views[viewName, schemaName]?.Script()
+                    : database.Views[viewName]?.Script();
+        }
+
+        /// <summary>
+        /// Script a stored procedure using SMO
+        /// </summary>
+        /// <param name="storedProcedureName">Stored Procedure name</param>
+        /// <param name="schemaName">Schema Name</param>
+        /// <returns>String collection of scripts</returns>
+        internal StringCollection GetStoredProcedureScripts(string viewName, string schemaName)
+        {
+            return (schemaName != null) ? database.StoredProcedures[viewName, schemaName]?.Script()
+                    : database.StoredProcedures[viewName]?.Script();
+        }
+
+        /// <summary>
+        /// Script a object using SMO and write to a file.
+        /// </summary>
+        /// <param name="sqlScriptGetter">Function that returns the SMO scripts for an object</param>
+        /// <param name="objectName">SQL object name</param>
+        /// <param name="schemaName">Schema name or null</param>
+        /// <param name="objectType">Type of SQL object</param>
+        /// <returns>Location object representing URI and range of the script file</returns>
+        internal Location[] GetSqlObjectDefinition(
+                Func<string, string, StringCollection> sqlScriptGetter, 
+                string objectName, 
+                string schemaName, 
+                string objectType) 
+        {
+            if (this.connectionInfo.SqlConnection != null)
+            {
+                StringCollection scripts = sqlScriptGetter(objectName, schemaName);
+                string tempFileName = (schemaName != null) ?  Path.Combine( tempPath, String.Format("{0}.{1}.sql", schemaName, objectName)) 
+                                                    : Path.Combine( tempPath, String.Format("{0}.sql", objectName));
+
+                if (scripts != null)
+                {
+                    int lineNumber = 0;
+                    using (StreamWriter scriptFile = new StreamWriter(File.Open(tempFileName, FileMode.Create, FileAccess.ReadWrite)))
+                    {
+                        
+                        foreach (string script in scripts)
+                        {
+                            if (script.IndexOf(String.Format("CREATE {0}", objectType), StringComparison.OrdinalIgnoreCase) >= 0)
```

Comment:
>String [](start = 47, length = 6)

please use "string" instead of "String" as that's the code guideline we follow. It's nice if we keep this consistent